sdks/java/extensions/jackson/src/test/java/org/apache/beam/sdk/extensions/jackson/JacksonTransformsTest.java

```diff
@@ -0,0 +1,242 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.extensions.jackson;
+
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import com.google.common.collect.Iterables;
+import java.io.Serializable;
+import java.util.Arrays;
+import java.util.List;
+import org.apache.beam.sdk.Pipeline;
+import org.apache.beam.sdk.coders.SerializableCoder;
+import org.apache.beam.sdk.coders.StringUtf8Coder;
+import org.apache.beam.sdk.testing.PAssert;
+import org.apache.beam.sdk.testing.TestPipeline;
+import org.apache.beam.sdk.transforms.Create;
+import org.apache.beam.sdk.values.PCollection;
+import org.junit.Rule;
+import org.junit.Test;
+
+/**
+ * Test Jackson transforms {@link ParseJsons} and {@link AsJsons}.
+ */
+public class JacksonTransformsTest {
+  private static final List<String> VALID_JSONS =
+      Arrays.asList(
+          "{\"myString\":\"abc\",\"myInt\":3}",
```

Comment:
You're right about JSON files. However, these transforms are for transforming a `PCollection` of JSONs to objects or a `PCollection` of objects to JSONs.

There were talks about doing a `JsonIO` too. But it felt like it just delegates to `TextIO`, and it didn't feel right.
However you raise a valid point that if a user wants to write the `PCollection` to a file, say using `TextIO` they would need to know to add the appropriate header and footer via `TextIO`s builder methods.

@dhalperi @eljefe6a Do you think it is worth reopening the possibility of adding a `JsonIO` in this PR as well?